skfem/assembly/basis/interior_basis.py

```diff
@@ -144,3 +144,13 @@ def interpfun(x):
             return w
 
         return interpfun
+
+    def with_element(self, elem: Element) -> 'InteriorBasis':
+
+        return type(self)(
+            self.mesh,
+            elem,
+            mapping=self.mapping,
+            quadrature=self.quadrature,
+            elements=self.tind,
+        )
```

Comment:
Were `InteriorBasis` a [`dataclass`](https://docs.python.org/3/library/dataclasses.html?highlight=dataclass#dataclasses.dataclass) (python>=3.7), this would be handled by [`replace`](https://docs.python.org/3/library/dataclasses.html?highlight=dataclass#dataclasses.replace): 
```python
        return replace(self, elem=elem)
```

Reply:
Will that also recompute `self.basis`? I thought `dataclass` is for those cases where you have only a trivial constructor. However, the main point of `Basis` is to cache this line: https://github.com/kinnala/scikit-fem/blob/master/skfem/assembly/basis/interior_basis.py#L67

Reply:
Would it be possible to not have `self.basis = ...` in constructor, but rather as a `property` which is cached and then invalidated during a call to `replace`?

Reply:
I really need to look into `dataclass` soon because Python 3.6 EOL is coming near and it becomes a viable option for us.